parallel-install/pkg/preinstaller/resourceparser.go

```diff
@@ -0,0 +1,99 @@
+package preinstaller
+
+import (
+	"encoding/json"
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+)
+
+// ResourceParser parses a resource from a given input.
+type ResourceParser interface {
+	// ParseUnstructuredResourceFrom given path and return it.
+	ParseUnstructuredResourceFrom(path string) (*unstructured.Unstructured, error)
+}
+
+// GenericResourceParser is a default implementation of ResourceParser.
+type GenericResourceParser struct{}
+
+// NewGenericResourceParser returns a new instance of GenericResourceParser.
+func NewGenericResourceParser() *GenericResourceParser {
```

Comment:
If a constructor is not doing anything on the struct, it is not needed. One can just instantiate the struct:
`grp := &preinstaller.GenericResourceParser{}`